src/server/plugins/attendee-auth-strategy.ts

```diff
@@ -0,0 +1,93 @@
+import { Request } from 'hapi'
+import { Logger } from 'pino'
+import { WebClient, UsersInfoResponse } from '@slack/client'
+import { PluginRegister } from '../../hapi.types'
+import { AttendeeModel } from '../models'
+
+type ValidateFuncCallback = (err: any, isValid?: boolean, credentials?: any) => void
+
+interface CustomOptions {
+  slack: WebClient,
+  password: string,
+}
+
+async function validateAttendeeUser(username: string, slack: WebClient, log: Logger) {
+  if (username.indexOf('@') > -1) {
+    log.info(`Finding attendee with email "${username}"...`)
+
+    // Username is an attendee email address
```

Comment:
But you could extract this as `validateAttendeeEmail` and then pull out the other part as `validateAttendeeSlackId` to make it obvious.